test/unit/custom_spec.js

```diff
@@ -107,3 +107,131 @@ describe("custom canvas rendering", function () {
       .catch(done.fail);
   });
 });
+
+describe("custom ownerDocument", function () {
+  const FontFace = globalThis.FontFace;
+
+  const checkFont = font => /g_d\d+_f1/.test(font.family);
+  const checkFontFaceRule = rule =>
+    /^@font-face {font-family:"g_d\d+_f1";src:/.test(rule);
+  const collectionMatches = (iterable, matcherFunction) =>
+    Array.from(iterable).find(matcherFunction) !== undefined;
+
+  beforeEach(() => {
+    globalThis.FontFace = function MockFontFace(name) {
+      this.family = name;
+    };
+  });
+
+  afterEach(() => {
+    globalThis.FontFace = FontFace;
+  });
+
+  function getMocks() {
+    const elements = [];
+    const createElement = name => {
+      let element =
+        typeof document !== "undefined" && document.createElement(name);
+      if (name === "style") {
+        element = {
+          tagName: name,
+          sheet: {
+            cssRules: [],
+            insertRule(rule) {
+              this.cssRules.push(rule);
+            },
+          },
+        };
+        Object.assign(element, {
+          remove() {
+            this.remove.called = true;
+          },
+        });
+      }
+      elements.push(element);
+      return element;
+    };
+    const ownerDocument = {
+      fonts: new Set(),
+      createElement,
+      documentElement: {
+        getElementsByTagName: () => [{ appendChild: () => {} }],
+      },
+    };
+
+    const CanvasFactory = isNodeJS
+      ? new NodeCanvasFactory()
+      : new DOMCanvasFactory({ ownerDocument });
+    return {
+      elements,
+      ownerDocument,
+      CanvasFactory,
+    };
+  }
+
+  it("should use given document for loading fonts (with Font Loading API)", async function () {
+    const { ownerDocument, elements, CanvasFactory } = getMocks();
+    const getDocumentParams = buildGetDocumentParams(
+      "TrueType_without_cmap.pdf",
+      {
+        disableFontFace: false,
+        ownerDocument,
+      }
+    );
+
+    const loadingTask = getDocument(getDocumentParams);
+    const doc = await loadingTask.promise;
+    const page = await doc.getPage(1);
+
+    const viewport = page.getViewport({ scale: 1 });
+    const canvasAndCtx = CanvasFactory.create(viewport.width, viewport.height);
+
+    await page.render({
+      canvasContext: canvasAndCtx.context,
+      viewport,
+    }).promise;
+
+    const style = elements.find(element => element.tagName === "style");
+    expect(style).toBeFalsy();
+    expect(ownerDocument.fonts.size).toBeGreaterThanOrEqual(1);
+    expect(collectionMatches(ownerDocument.fonts, checkFont)).toBeTruthy();
+    await doc.destroy();
+    await loadingTask.destroy();
+    CanvasFactory.destroy(canvasAndCtx);
+    expect(ownerDocument.fonts.size).toBe(0);
+  });
+
+  it("should use given document for loading fonts (with CSS rules)", async function () {
+    const { ownerDocument, elements, CanvasFactory } = getMocks();
+    ownerDocument.fonts = null;
```

Comment:
Is this line necessary? You're creating new mocks in each test, so this will be an empty set by default, but since you're only checking the style tags in this test that shouldn't matter, right?

Reply:
that line is what triggers the font loader to fallback. If the document has a truthy value at that key it tries to use that api.

Reply:
Ah, I didn't know that. Thanks!